### PR TITLE
Updates to authorization flow.

### PR DESCRIPTION
**Pull request by mahima-desetty**
This is an ongoing effort to update the code snippets with the auth flow from the Java quickstart instead of the Application Default Credentials. 

I've updated two classes in this PR to get feedback/comments early before I update the rest of them.

### CODE REVIEW

File: classroom/snippets/src/main/java/AddAliasToCourse.java [hunk 1]
```diff
@@ -37,21 +36,20 @@ public class AddAliasToCourse {
...
+    tokens/ folder. */
+    final List<String> SCOPES = Collections.singletonList(ClassroomScopes.CLASSROOM_COURSES);
 
-    // Create the classroom API client.
-    Classroom service = new Classroom.Builder(new NetHttpTransport(),
-        GsonFactory.getDefaultInstance(),
-        requestInitializer)
-        .setApplicationName("Classroom samples")
-        .build();
+    // Create the classroom API client
```
Comment (AndrewMBurke) [inline classroom/snippets/src/main/java/AddAliasToCourse.java]: Should be `// Create the Classroom API client.`.

File: classroom/snippets/src/main/java/CreateCourse.java [hunk 2]
```diff
@@ -36,21 +35,20 @@ public class CreateCourse {
...
-    GoogleCredentials credentials = GoogleCredentials.getApplicationDefault()
-        .createScoped(Collections.singleton(ClassroomScopes.CLASSROOM_COURSES));
-    HttpRequestInitializer requestInitializer = new HttpCredentialsAdapter(
-        credentials);
+  public static Course createCourse() throws Exception {
+    /* Scopes required by this API call. If modifying these scopes, delete your previously saved
+    tokens/ folder. */
+    final List<String> SCOPES = Collections.singletonList(ClassroomScopes.CLASSROOM_COURSES);
 
     // Create the classroom API client
```
Comment (AndrewMBurke) [inline classroom/snippets/src/main/java/CreateCourse.java]: Should be `// Create the Classroom API client.`.

File: classroom/snippets/src/test/java/TestAddAliasToCourse.java [hunk 3]
```diff
@@ -22,7 +22,7 @@
 public class TestAddAliasToCourse extends BaseTest {
 
   @Test
-  public void testAddCourseAlias() throws IOException {
+  public void testAddCourseAlias() throws Exception {
```
Comment (AndrewMBurke) [inline classroom/snippets/src/test/java/TestAddAliasToCourse.java]: Why the change to the generic `Exception`?

Comment (mahima-desetty) [reply classroom/snippets/src/test/java/TestAddAliasToCourse.java]: GoogleNetHttpTransport throws a GeneralSecurityException so the method could throw either error. I changed it to Exception since IOException and GeneralSecurityException belong to the Exception class.

**Review by DavidScales — APPROVED** [review]

File: classroom/snippets/src/test/java/TestAddAliasToCourse.java (hunk 3 above)
Comment (AndrewMBurke) [reply classroom/snippets/src/test/java/TestAddAliasToCourse.java]: If we know the specific exceptions that may occur (and the number is reasonable), IMO we should show them with explanation about why they might occur. This doesn't seem unreasonable:

`public void testAddCourseAlias() throws GeneralSecurityException, IOException {`

Comment (mahima-desetty) [reply classroom/snippets/src/test/java/TestAddAliasToCourse.java]: Updated the method signature to include specific errors.

File: classroom/snippets/src/main/java/AddAliasToCourse.java [hunk 4]
```diff
@@ -12,46 +12,46 @@
...
 
 /* Class to demonstrate the use of Classroom Create Alias API. */
 public class AddAliasToCourse {
+
   /**
    * Add an alias on an existing course.
    *
    * @param courseId - id of the course to add an alias to.
    * @return - newly created course alias.
    * @throws IOException - if credentials file not found.
```
Comment (AndrewMBurke) [inline classroom/snippets/src/main/java/AddAliasToCourse.java:41]: Need an annotation for the second exception type, `@throws GeneralSecurityException`.

File: classroom/snippets/src/main/java/CreateCourse.java [hunk 5]
```diff
@@ -12,45 +12,44 @@
...
+import java.util.List;
 
 /* Class to demonstrate the use of Classroom Create Course API */
 public class CreateCourse {
+
   /**
    * Creates a course
    *
    * @return newly created course
    * @throws IOException - if credentials file not found.
```
Comment (AndrewMBurke) [inline classroom/snippets/src/main/java/CreateCourse.java:40]: Same comment as above re: annotation for the other exception type.

**Review by AndrewMBurke — APPROVED** [review]
LGTM with adding the two missing annotations.

File: classroom/snippets/src/test/java/BaseTest.java [hunk 6]
```diff
@@ -38,10 +40,10 @@ public class BaseTest {
    * @return an authorized Classroom client service
    * @throws IOException - if credentials file not found.
    */
-  protected Classroom buildService() throws Exception {
+  protected Classroom buildService(ArrayList<String> SCOPES) throws GeneralSecurityException, IOException {
```
Comment (DavidScales) [inline classroom/snippets/src/test/java/BaseTest.java:43]: I'm probably missing a basic concept but why do we even need to build the service in the tests? Doesn't the class under test already build the service? Are the credentials scoped to certain files or something?

File: classroom/snippets/src/test/java/BaseTest.java [hunk 7]
```diff
@@ -38,10 +40,10 @@ public class BaseTest {
...
     /* Scopes required by this API call. If modifying these scopes, delete your previously saved
     tokens/ folder. */
-    final List<String> SCOPES = Collections.singletonList(ClassroomScopes.CLASSROOM_COURSES);
+    SCOPES.add(ClassroomScopes.CLASSROOM_COURSES);
```
Comment (DavidScales) [inline classroom/snippets/src/test/java/BaseTest.java:46]: Do you know why we were originally adding the `courses` scopes manually in the tests? In the new pattern it looks like you're adding the scopes from the classes under test with `setup` so I wonder if we really need this?

File: classroom/snippets/src/main/java/CreateCourse.java [hunk 8]
```diff
@@ -65,7 +64,7 @@ public static Course createCourse() throws Exception {
               + "to be excited!")
           .setRoom("301")
           .setOwnerId("me")
-          .setCourseState("PROVISIONED");
+          .setCourseState("ACTIVE");
```
Comment (DavidScales) [inline classroom/snippets/src/main/java/CreateCourse.java:68]: I've seen other issues developers have with the understanding course states, so it might be worthwhile to leave a comment here that calls out what ACTIVE means and links to the appropriate documentation

File: classroom/snippets/src/main/java/GetTopic.java [hunk 9]
```diff
@@ -63,12 +66,7 @@ public static Topic getTopic(String courseId, String topicId) throws IOException
       System.out.printf("Topic '%s' found.\n", topic.getName());
     } catch (GoogleJsonResponseException e) {
       //TODO (developer) - handle error appropriately
-      GoogleJsonError error = e.getDetails();
```
Comment (DavidScales) [inline classroom/snippets/src/main/java/GetTopic.java:66]: why did you modify how these topic errors are handled? I think the Topic error handling sections are different from each other now

File: classroom/snippets/src/test/java/TestAddStudent.java [hunk 10]
```diff
@@ -12,22 +12,26 @@
...
-    deleteCourse(course.getId());
+  public void testAddStudent() throws GeneralSecurityException, IOException {
+    // Include the scopes required to run the code example for testing purposes.
+    setup(AddStudent.SCOPES);
+    Student student = AddStudent.addStudent(testCourse.getId(), testCourse.getEnrollmentCode(),
+        this.studentId);
     Assert.assertNotNull("Student not returned.", student);
-    Assert.assertEquals("Student added to wrong course.", course.getId(),
+    Assert.assertNotNull("Course not returned.", student.getCourseId());
+    Assert.assertEquals("Student added to wrong course.", testCourse.getId(),
```
Comment (DavidScales) [inline classroom/snippets/src/test/java/TestAddStudent.java:34]: this test doesn't look right; shouldn't `assertEquals` have two values?

**Review by DavidScales — COMMENTED** [review]
The need to remove the `@before` and have repeated code in each test is not ideal but I can't think of a better way myself. We could have the base test request ALL scopes, but I think that would de-sync the tests too much from the application code (i.e., tests would pass even if the code snippets had the wrong scopes).

File: classroom/snippets/src/main/java/AddAliasToCourse.java (hunk 4 above)
Comment (mahima-desetty) [reply classroom/snippets/src/main/java/AddAliasToCourse.java:41]: Done

File: classroom/snippets/src/main/java/CreateCourse.java (hunk 5 above)
Comment (mahima-desetty) [reply classroom/snippets/src/main/java/CreateCourse.java:40]: Done

File: classroom/snippets/src/test/java/BaseTest.java (hunk 6 above)
Comment (mahima-desetty) [reply classroom/snippets/src/test/java/BaseTest.java:43]: The BaseTest is responsible for creating a course that all of the other tests can modify which is why it has to run first and build the service. The service that is created is only scoped to the BaseTest class so another service needs to be created in the other Java classes in order for the methods to be called. But, the credentials are only created once (when the service creation is first run), and any subsequent API calls will use the credentials stored in the tokens/ directory if it exists.

File: classroom/snippets/src/main/java/CreateCourse.java (hunk 8 above)
Comment (mahima-desetty) [reply classroom/snippets/src/main/java/CreateCourse.java:68]: Done

File: classroom/snippets/src/main/java/GetTopic.java (hunk 9 above)
Comment (mahima-desetty) [reply classroom/snippets/src/main/java/GetTopic.java:66]: The assertThrows check in TestDeleteTopic requires a GetTopic call to throw a GoogleJsonResponseException. In the previous code, that specific exception was being caught without throwing the error. I've modified this to throw the error, but re-added the previous code as well so that the error handling sections are similar.

There is a TODO in there for developers to be aware that they must implement their own error handling, so I'm not too concerned if the error handling differs between classes or doesn't catch every case. The main purpose of the code snippets are to show how to make various method calls in Java.

File: classroom/snippets/src/test/java/BaseTest.java (hunk 7 above)
Comment (mahima-desetty) [reply classroom/snippets/src/test/java/BaseTest.java:46]: The old scopes were to be included so the application default credentials could set up the service in BaseTest. This is needed in the BaseTest because the courses scope is required when calling CreateCourse in the setup() method. The scopes in the classes under the test are only adding the scopes specific to that API call. So, for example, calling TestDeleteTopics requires the topics scope to call DeleteTopic and the courses scope to create the testCourse in BaseTest.

File: classroom/snippets/src/test/java/TestAddStudent.java (hunk 10 above)
Comment (mahima-desetty) [reply classroom/snippets/src/test/java/TestAddStudent.java:34]: This is comparing the testCourse `id` with the student object `courseId`.
